system-tests/tests/src/testFixtures/java/org/eclipse/dataspaceconnector/system/tests/utils/TransferSimulationUtils.java

```diff
@@ -69,10 +70,30 @@ private TransferSimulationUtils() {
      * @param requestFactory Factory for creating transfer request payloads.
      */
     public static ChainBuilder contractNegotiationAndTransfer(String providerUrl, TransferRequestFactory requestFactory) {
-        return startContractAgreement(providerUrl)
+        return getOffer(providerUrl)
+                .exec(negotiateContract(providerUrl))
                 .exec(waitForContractAgreement())
                 .exec(startTransfer(providerUrl, requestFactory))
-                .exec(waitForTransferCompletion());
+                .exec(waitForTransferCompletion())
+                .exec(deprovision());
```

Comment:
including `deprovision` here can cause a challenge that if in the test we want to assert content of the copied blob after transfer is complete then that will not be possible as provisioned container is deprovisioned during the gatling run.

Reply:
True - however I don't see that adding this introduces a challenge:
- this assertion is not used currently, we can adapt if/when it is needed
- this assertion would logically belong within the gatling chain (i.e. before deprovisioning and not outside)
- it would only affect a test using `managedResources=true` 
- we can always add a setting to optionally skip deprovision

Reply:
@cpeeyush @algattik is there an intention to include an assertion for confirming the content of the copied blob at this time?  My perspective is that it would make sense to but should be part of a different story

Reply:
@algattik In our system tests we do assert the destination blob exists and its contents. 

- https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/ead9b4d7d4f279a4954a5272630645795c842822/system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java#L135
- https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/ead9b4d7d4f279a4954a5272630645795c842822/system-tests/azure-data-factory-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/AzureDataFactoryTransferIntegrationTest.java#L164


So these assertions will fail because of this change ? correct ?

Reply:
Fixed by making output check a part of the process